src/core/Akka.Cluster.Tests/MemberOrderingModelBasedTests.cs

```diff
@@ -0,0 +1,250 @@
+            if (addressCount != distinctCount)
+                return true.Classify(false, "all distinct").Classify(false, "empty set");
+
+            // use the sort to help us short-circuit a comparison problem
+            var sortedAddresses = addresses.OrderBy(x => x, Member.AddressOrdering).ToList();
+
+            var a1 = sortedAddresses.First();
+            foreach (var a2 in sortedAddresses.Skip(1))
+            {
+                if (Member.AddressOrdering.Compare(a1, a2) == 0)
+                    return false.Classify(true, "all distinct").Classify(false, "empty set")
+                        .Label($"Expected {a1} and {a2} to compare as different, but did not. Compared as same");
+                a1 = a2; // next member
+            }
+
+            return true.Classify(true, "all distinct").Classify(false, "empty set"); ;
+        }
+    }
+
+    public class MembershipMachine : Machine<MembershipState, MembershipModel>
+    {
+        public MembershipMachine()
+        {
+            // register the custom generators to make testing easier
+            Arb.Register<ClusterGenerators>();
+        }
+
+        public override Gen<Operation<MembershipState, MembershipModel>> Next(MembershipModel model)
+        {
+            if (model.AllMembers.Count == 0)
+                return AddNewMember.Generator();
+            return Gen.OneOf(ChangeMemberStatus.Generator(model.AllMembers.Keys), AddNewMember.Generator());
+        }
+
+        public override Arbitrary<Setup<MembershipState, MembershipModel>> Setup => Arb.From(Arb.Generate<MembershipSetup>()
+            .Select(x => (Setup<MembershipState, MembershipModel>)x)); // compiler ceremony :(
+
+        #region Setup
+
+        public class MembershipSetup : Setup<MembershipState, MembershipModel>
+        {
+            private readonly Member[] _members;
+
+            public MembershipSetup(UniqueAddress[] addresses)
+            {
+                // filter out any duplicates
+                _members =
+                    addresses.Distinct()
+                        .Select(x => new Member(x, int.MaxValue, MemberStatus.Up, ImmutableHashSet<string>.Empty))
+                        .ToArray();
+            }
+
+            public override MembershipState Actual()
+            {
+                return new MembershipState() { Members = ImmutableSortedSet<Member>.Empty.Union(_members) };
+            }
+
+            public override MembershipModel Model()
+            {
+                return new MembershipModel(_members.ToImmutableDictionary(x => x.Address, x => x));
+            }
+
+            public override string ToString()
+            {
+                return $"{GetType()}(Members = [{string.Join(",", _members.Select(x => x.ToString()))}]";
+            }
+        }
+
+        #endregion
+
+        #region Operations
+
+        public class ChangeMemberStatus : Operation<MembershipState, MembershipModel>
```

Comment:
Verifies that if we change the status of a current member of the cluster to a new `MembershipStatus`, as long as that transition is valid the new instance of the `Member` will be merged correctly into the `ImmutableSortedSet<Member>` collection.